exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionSenderRootExec.java

```diff
@@ -98,10 +100,12 @@ public int metricId() {
 
   public PartitionSenderRootExec(FragmentContext context,
                                  RecordBatch incoming,
-                                 HashPartitionSender operator) throws OutOfMemoryException {
+                                 HashPartitionSender operator,
+                                 boolean closeIncoming) throws OutOfMemoryException {
```

Comment:
Introduce back constructor without `closeIncoming` argument and call this constructor from it with `closeIncoming` set to `false`.